src/test/java/com/hotels/jasvorno/JasvornoConverterTest.java

```diff
@@ -109,6 +121,150 @@ public void unknownWithOptional() throws Exception {
   }
 
   @Test
+  public void happyWithoutModel() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder
+        .builder()
+        .record("record")
+        .fields()
+        .requiredLong("id")
+        .requiredString("name")
+        .endRecord();
+    String json = "{ \"id\": 1, \"name\": \"bob\" }";
+    JsonNode datum = mapper.readTree(json);
+    JasvornoConverter.convertToAvro(datum, schema, UndeclaredFieldBehaviour.IGNORE);
+  }
+
+  @Test
+  public void nullDatumValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder.builder().record("record").fields().requiredLong("id").endRecord();
+    Object obj = JasvornoConverter.convertToAvro(null, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj == null);
+  }
+
+  @Test
+  public void nullDatumValueWithModel() throws Exception {
+    Schema schema = SchemaBuilder
+        .builder()
+        .record("record")
+        .fields()
+        .requiredLong("id")
+        .requiredString("name")
+        .endRecord();
+    Object obj = JasvornoConverter.convertToAvro(model, null, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj == null);
+  }
+
+  @Test(expected = JasvornoConverterException.class)
+  public void datumWithNullValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder.builder().record("record").fields().requiredString("name").endRecord();
+    ObjectMapper mapper = new ObjectMapper();
+    ObjectNode dataTable = mapper.createObjectNode();
+    JsonNode datum = dataTable.putNull("name");
+    JasvornoConverter.convertToAvro(model, datum, schema, UndeclaredFieldBehaviour.IGNORE);
+  }
+
+  @Test
+  public void datumWithNoValue() throws JsonProcessingException, IOException {
+    Schema schema = SchemaBuilder
+        .record("record")
+        .fields()
+        .name("name")
+        .type()
+        .stringType()
+        .stringDefault("x")
+        .endRecord();
+    JsonNode datum = mapper.readTree("{}");
+    Object obj = JasvornoConverter.convertToAvro(model, datum, schema, UndeclaredFieldBehaviour.IGNORE);
+    assertTrue(obj instanceof IndexedRecord);
+    IndexedRecord record = (IndexedRecord) obj;
+    assertTrue(record.get(0).toString().equals("x"));
```

Comment:
assertThat(record.get(0).toString(), is("x")); -> this is clearer and also gives you better error messages along the lines of "Expected 'x' but got 'foo'" while something failing assertTrue will just say something line "wasn't true" which makes it harder to debug.